src/lang/processor.rs

```diff
@@ -0,0 +1,1347 @@
+            })?
+    } else {
+        pair.clone()
+    };
+
+    let raw_content = inner_pair.as_str();
+    let mut result = String::with_capacity(raw_content.len());
+    let mut chars = raw_content.chars().peekable();
+
+    while let Some(c) = chars.next() {
+        if c == '\\' {
+            match chars.next() {
+                Some('"') => result.push('"'),
+                Some('\\') => result.push('\\'),
+                Some('/') => result.push('/'),
+                Some('b') => result.push('\x08'),
+                Some('f') => result.push('\x0C'),
+                Some('n') => result.push('\n'),
+                Some('r') => result.push('\r'),
+                Some('t') => result.push('\t'),
+                Some('u') => {
+                    let mut hex_code = String::with_capacity(4);
+                    for _ in 0..4 {
+                        hex_code.push(chars.next().ok_or_else(|| {
+                            ProcessorError::InvalidLiteralFormat {
+                                kind: "unicode escape".to_string(),
+                                value: format!("\\u{}... (incomplete)", hex_code),
+                                span: Some(get_span(&inner_pair)),
+                            }
+                        })?);
+                    }
+                    let char_code = u32::from_str_radix(&hex_code, 16).map_err(|_| {
+                        ProcessorError::InvalidLiteralFormat {
+                            kind: "unicode escape".to_string(),
+                            value: format!("\\u{}", hex_code),
+                            span: Some(get_span(&inner_pair)),
+                        }
+                    })?;
+                    result.push(std::char::from_u32(char_code).ok_or_else(|| {
+                        ProcessorError::InvalidLiteralFormat {
+                            kind: "unicode escape (invalid code point)".to_string(),
+                            value: format!("\\u{}", hex_code),
+                            span: Some(get_span(&inner_pair)),
+                        }
+                    })?);
+                }
+                Some(other) => {
+                    return Err(ProcessorError::InvalidLiteralFormat {
+                        kind: "escape sequence".to_string(),
+                        value: format!("\\{}", other),
+                        span: Some(get_span(&inner_pair)),
+                    })
+                }
+                None => {
+                    return Err(ProcessorError::InvalidLiteralFormat {
+                        kind: "escape sequence".to_string(),
+                        value: "\\ (ends with escape)".to_string(),
+                        span: Some(get_span(&inner_pair)),
+                    })
+                }
+            }
+        } else {
+            result.push(c);
+        }
+    }
+    Ok(result)
+}
+
+fn parse_hex_str_to_raw_value(hex_str: &str) -> Result<middleware::RawValue, ProcessorError> {
```

Comment:
I just realized we already had this functionality here: https://github.com/0xPARC/pod2/blob/d3fef8392e110121e0b4284995d696cc873a64f9/src/middleware/basetypes.rs#L209-L218

It's used to deserialize a `Hash` from a hex string, and internally it creates the 4 field elements, which is the same we need to get a RawValue.  Then you can do `RawValue::from(h: Hash)`.

The encoding is different though, and it doesn't use a `0x` prefix.
The encoding uses little endian, and taking a little endian hex encoded value and adding a `0x` prefix is confusing from my point of view.